### DISCUSSION TITLE
How do Safeboot update

### DISCUSSION THREAD

**Discussion by DittelHome** [arendst/Tasmota#21353] [Q&A] [answered]
Hi Community,
my ESP32 c3 shows version 13.4 (after flashing my own build, which i generated with gitpod), while safeboot shows version 13.3.
How can i update the safeboot partition ?

Documentation says:
```
otaurl http://ota.tasmota.com/tasmota32/release-13.4.0/tasmota32.bin
upload 2
```

Should i type:
`http://ota.tasmota.com/tasmota32/release-13.4.0/tasmota32c3-safeboot.bin`
in the OTA update field ?

What i have to do with upload 2?
Sorry maybe stupid questions....

regards Klaus

**Comment by DittelHome** [comment, marked as answer]
In fact, it was a stupid question ...
I have typed in (at console) the follwing commands:
```
otaurl http://ota.tasmota.com/tasmota32/release-13.4.0/tasmota32c3-safeboot.bin
upload 2

```

Now safeboot is on 13.4 ..

Sorry for the circumstances ..

**Reply by sfromis** [reply]
To upgrade safeboot, the general intention is **not** that you point to the specific URL for the safeboot variant, but to the corresponding regular binary. Then the upgrade of safeboot will modify the URL to match. This allows the normal flow with precompiled binaries to include safeboot upgrades too, without needing to update `OtaUrl`. Hence the documentation is correct, except that you have to match the ESP32 variant architecture.

The situation is a bit more mixed if you are building custom binaries. While you still can use the regular URL (instead of safeboot) for the purpose of getting safeboot upgraded, it does not provide the straightforward flow you get with the precompiled binaries.

What I'm doing with my own compiles is to have a local OTA server to serve the resulting binaries, and this server also provides the safeboot binaries when requested, allowing me to have the straightforward upgrade flow (including with safeboot) as when using precompiled binaries from the Tasmota OTA server.

**Reply by alligator-asl** [reply]
How van I determine the currently installed safeboot version?

**Reply by sfromis** [reply]
The command `Restart 3` can be used to switch between the safeboot and normal OTA partition. While switched, you can check the Tasmota version by the usual means.

If you want to upgrade safeboot, use the command `Upgrade 2` while running in the normal OTA partition.
